### ISSUE TITLE
Additional granularity for types [.Alert, .Sound, .Badge] in UIUserNotificationSettings

### ISSUE THREAD

**Issue by mathiasgithub** [delba/Permission#23]
Is it possible to check for the additional types as in the title?

**Comment by delba** [comment]
It's currently not possible. The notifications permission is considered .Authorized if any notification type is granted.

It's not on the roadmap either but I would happily accept a PR that implements it nicely. Do you feel like doing it?

**Comment by delba** [comment]
I take it as a "no" 😃 

I'll ping you if/when I implement it. Thanks for the suggestion!

**Comment by mathiasgithub** [comment]
Sorry for late answer!

I'll take a look at it, but I'm not sure if I can fit it in the schedule
unfortunately

I appreciate the work you do!

2016-04-22 5:51 GMT+02:00 Damien notifications@github.com:

**Comment by delba** [comment]
Hi @mathiasgithub,

I added additional granularity for notification types in https://github.com/delba/Permission/commit/25ff83088680b8f1ca8a121fa3dd7a97bcd9b0d4 and I'd be interested in hearing your thoughts.

By default, the types are `[.Badge, .Sound, .Alert]` and the categories are `nil`:

``` swift
let notifications: Permission = .Notifications
```

And you can also specify the types and categories:

``` swift
let notifications: Permission = .Notifications(types: [.Alert], categories: nil)
```

**Comment by delba** [comment]
Changes included in `v1.5`
